multi_physics/QED/QED_table_generator/table_generator.cpp

```diff
@@ -198,71 +298,440 @@ void generate_quantum_sync_photem_table(
+        << OPT_PRECISION_SINGLE << ", "
+        << OPT_PRECISION_SINGLE_WITH_DOUBLE_COMP << ".\n"
+        << std::setw(MAX_CMD_SIZE+3) << "" << OPT_PRECISION_SINGLE_WITH_DOUBLE_COMP
+        << " means that calculations are carried out in double precision\n"
+        << std::setw(MAX_CMD_SIZE+3) << ""
+        << "even if the table is in single precision. " << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_CHI_MIN <<
+        " : sets minimum chi parameter (real number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_CHI_MAX <<
+        " : sets maximum chi parameter (real number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_FRAC_MIN <<
+        " : sets minimum frac parameter (real number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_CHI_SIZE <<
+        " : sets number of chi parameters (integer number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_FRAC_SIZE <<
+        " : sets number of frac parameters (integer number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_FILENAME <<
+        " : sets output file name (string)" << std::endl;
+
+    std::cout << " * Optional parameters have default values." << std::endl;
+
+    std::cout << " ____________________________" << std::endl;
+}
+
+/**
+* Prints an error message
+*
+* @param[in] str the error message
+*/
+void print_error(const std::string& str)
+{
+    std::cout << "\n [ERROR!] " << str << "\n" << std::endl;
+}
+
+/**
+* Checks the number of command line arguments and exits if necessary
+*
+* @param[in] argc the number of command line arguments
+*/
+void check_argc(int argc)
+{
+    // We must pass an even number of arguments (and at least 1 argument),
+    // thus argc must be odd and > 2
+    if (argc%2 != 1 || argc < 2){
+        print_error("The number of arguments is incorrect!");
+        print_help_message();
+        exit(EXIT_FAILURE);
+    }
+}
+
+/**
+* Handles the special case in which the user asks to print the help message
+*
+* @param[in] argc the number of command line arguments
+* @param[in] argv the command line arguments
+*/
+void handle_help_argument(int argc, char** argv)
+{
+    if(argc < 2)
+        return;
+
+    const auto arg = std::string{argv[1]};
+    if(arg.compare(CMD_HELP_S) == 0 || arg.compare(CMD_HELP_L) == 0){
+        print_help_message();
+        exit(EXIT_SUCCESS);
+    }
+}
+
+/**
+* Reads command line arguments in (command, argument) pairs
+*
+* @param[in] argc the number of command line arguments
+* @param[in] argv the command line arguments
+* @return the command line arguments in a (command --> argument) map
+*/
+std::map<std::string, std::string> read_arg_pairs (int argc, char** argv)
+{
+    check_argc(argc);
+
+    auto arg_map = std::map<std::string, std::string>{};
+
+    for (int i = 1; i < argc; i += 2){
+        arg_map[argv[i]] = argv[i+1];
```

Comment:
You can potentially add a construct like `if ( arg_map.find(argv[i]) == arg_map.end() )` to ensure that the same input argument has not been used twice and exit if it has.

Reply:
It's a very good idea!

Reply:
Done!
```
./table_generator --table_type breit_wheeler --precision single --name bw_s --name dummy

 [ERROR!] Command line argument '--name' is duplicated!

```